### PR TITLE
Set buffering on all input file descriptors

### PR DESCRIPTION
**Pull request by josuah**
Note that there is no fclose(), as there is no close() (rely on the operating system to close all file descriptors and free memory associated with the buffer).

The change is: `read()` -> `fread()`, `(int)fd` -> `(FILE *)fp` (which contains an extra buffer) with also `fdopen()` on the already open file descriptors to add the buffering.

### CODE REVIEW

File: jjd.c [hunk 1]
```diff
@@ -77,15 +77,15 @@ dial(const char *host, const char *port)
 }
 
 static int
-read_line(int fd, char *buf, size_t bufsize)
+read_line(FILE *fp, char *buf, size_t bufsize)
 {
-	char c;
+	int c;
```
Comment (josuah) [inline jjd.c:82]: Note here that `fgetc()` returns an int instead of a char, for having room for `EOF`

**Comment by aaronNGi** [comment]
Isn't this exactly the same thing as we already do? This PR still reads one char at a time while the goal would be to read as many chars as possible in one `read(3)` call, limited by what's available on the file descriptor.

**Comment by josuah** [comment]
It reads one character at a time, but this is `fgetc()` call instead of `read()`, which has a layer of buffering:

On the first time, fgetc() calls `read(fp->fd, fp->buffer, bufsize)`, which reads `bufsize` at a time, and store the amount of byte read in `fp->readsize`, and the position in `fp->pos`*.

On the next times, fgetc() first check if there are bytes not consumed in `fp->buffer`, and if so, returns `fp->buffer[fp->pos]`, shift pos (fp->pos++) and that is it: no read() performed until there is nothing left (`fp->pos == fp->readsize`).

So while looking like a per-byte read(), it is really a bufsize read(), and a per-byte fgetc().

This permits to work with the convenience of `fgetc()`, but the performance of `read(..., bufsize)`.

*this was an example, as there is no standard on what there is in the `FILE` structure: every libc (glibc, BSD libc, newlib, musl libc...) is free to make it however it wants, as long as the `<stdio.h>` functions behavior are respected.

**Comment by aaronNGi** [comment]
Got it, thanks for the explanation.

**Pull request merged**